File: http/src/main/java/webconnect/com/webconnect/di/IProperties.java
package webconnect.com.webconnect.di;

import android.app.Dialog;
import android.support.annotation.NonNull;

import java.util.Map;

import webconnect.com.webconnect.listener.AnalyticsListener;
import webconnect.com.webconnect.listener.OnWebCallback;

/**
 * Created by amit on 23/9/17.
 */

public interface IProperties<T> {

    T baseUrl(@NonNull String url);

    T headerParam(@NonNull Map<String, String> headerParam);

    T queryParam(@NonNull Map<String, String> queryParam);

    T callback(@NonNull OnWebCallback callback);

    T analyticsListener(@NonNull AnalyticsListener callback);

    T callback(@NonNull OnWebCallback callback,
               @NonNull Class<?> success, @NonNull Class<?> error);

    T taskId(int taskId);

    T timeOut(long connectTimeOut, long readTimeOut);

    T progressDialog(Dialog dialog);

    T cache(boolean isCache);

    void connect();
}

File: http/src/main/java/webconnect/com/webconnect/di/IProperties.kt
package webconnect.com.webconnect.di

import android.app.Dialog
import webconnect.com.webconnect.QueryMap
import webconnect.com.webconnect.listener.AnalyticsListener

/**
 * Created by amit on 23/9/17.
 */

interface IProperties<T> {

    fun baseUrl(url: String): T

    fun headerParam(headerParam: Map<String, String>): T

    fun queryParam(queryParam: QueryMap<String, String>): T

    fun analyticsListener(callback: AnalyticsListener): T

    fun taskId(taskId: Int): T

    fun timeOut(connectTimeOut: Long, readTimeOut: Long): T

    fun cache(isCache: Boolean): T

    fun connect()
}
